projects/vdk-plugins/vdk-trino/src/vdk/plugin/trino/ingest_to_trino.py

```diff
@@ -57,27 +62,107 @@ def _ingest_payload(
 
         http.client._MAXLINE = 6553600
 
-        query, fields = self._create_query_and_fields(
+        query, fields, types = self._create_query_and_fields(
             destination_table, len(payload), cur
         )
 
         try:
-            params = tuple(obj[field] for obj in payload for field in fields)
+            payload = self.__lowercase_keys_in_payload(payload)
+            params = self.__get_values_for_fields(payload, fields, types)
             cur.execute(query, params)
             cur.fetchall()
             log.debug("Payload was ingested.")
         except Exception as e:
             errors.log_and_rethrow(
                 errors.find_whom_to_blame_from_exception(e),
                 log,
-                "Failed to sent payload",
+                f"Failed to sent payload into table {destination_table}",
                 "Unknown error. Error message was : " + str(e),
                 "Will not be able to send the payload for ingestion",
                 "See error message for help ",
                 e,
                 wrap_in_vdk_error=True,
             )
 
+    @staticmethod
+    def __to_bool(value: str) -> bool:
+        if value == "true" or value == "True":
+            return True
+        if value == "false" or value == "False":
+            return False
+        raise ValueError("bool cast accept only True/true/False/false values.")
+
+    def __cast(self, key: str, new_value: Any, value_type: Type) -> Any:
+        try:
+            log.debug(f"Cast {key} to type {value_type}")
+            if value_type == bool:
+                return self.__to_bool(new_value)
+            if value_type == float:
+                value_with_float_type = float(new_value)
```

Comment:
Shouldn't there be a nested try-except here, in case `new_value` cannot be safely converted to float?

Reply:
It is covered by line 108

Reply:
So it is assumed that if the value cannot be converted to float, then everything up until line 108 should be skipped, correct?

Reply:
yes. Do you see any issue with that?

Reply:
Nope, just got a bit confused by the math.isnan check that follows, and thought that value_with_float_type is expected to have value after the conversion to float. However, I see why failing at the float conversion is better than handling the exception and checking the value after that.